scripts/gha/build_desktop.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Assuming pre-requisites are in place (from running 
+`scripts/gha/install_prereqs_desktop.py`), this builds the firebase cpp sdk.
+
+It does the following,
+- install desktop platform specific cpp dependencies via vcpkg
+- cmake configure
+- cmake build
+
+Usage examples:
+# Build all targets with x64 architecture, Release config and 8 cpu cores
+python scripts/gha/build_desktop.py --arch x64 --config Release -j 8
+
+# Build all targets with default options and also build unit tests
+python scripts/gha/build_desktop.py --build_tests --arch x64
+
+# Build only firebase_app and firebase_auth 
+python scripts/gha/build_desktop.py --target firebase_app firebase_auth
+
+"""
+
+import argparse
```

Comment:
I think we should land on absl.flags for all arg parsing in all our scripts. If you want to do this in a follow-up PR that's fine.

Reply:
Yup! Will do it in a follow up PR.